arelle/Cntlr.py

```diff
@@ -153,10 +154,10 @@ def __init__(
         }
         self.hasWin32gui = False
         self.hasGui = hasGui
-        self.hasFileSystem = True # no file system on Google App Engine servers
-        self.isGAE = False
-        self.isCGI = False
-        self.systemWordSize = int(round(math.log(sys.maxsize, 2)) + 1) # e.g., 32 or 64
+        self.hasFileSystem = hasFileSystem() # no file system on Google App Engine servers
+        self.isGAE = isGAE()
+        self.isCGI = isCGI()
+        self.systemWordSize = getSystemWordSize()  # e.g., 32 or 64
```

Comment:
I'd move the `self.isMac` and `self.isMSW` assignment up here and use those attributes below instead of `if sys.platform == "darwin":` etc.
```python
        platformOS = PlatformOS.getPlatformOS()
        self.isMac = platformOS == PlatformOS.MACOS
        self.isMSW = platformOS == PlatformOS.WINDOWS

        self.systemWordSize = getSystemWordSize()  # e.g., 32 or 64

        # ...
        if self.isMac:
            # ...
        elif self.isMSW:
            # ...
        else:
            # ...
```